tests/pgautofailover_utils.py

```diff
@@ -594,6 +610,71 @@ def enable_ssl(self, sslMode=None, sslSelfSigned=None,
         command = PGAutoCtl(self, argv=ssl_args)
         out, err = command.execute("enable ssl")
 
+    def check_ssl(self, ssl, sslmode, monitor=False, primary=False):
+        """
+        Checks if ssl settings match how the node is set up
+        """
+        key = self.sslServerKey
+        crt = self.sslServerCert
+        crl = None
+        rootCert = self.sslCAFile
+        if self.sslSelfSigned:
+            key = os.path.join(self.datadir, "server.key")
+            crt = os.path.join(self.datadir, "server.crt")
+
+        eq_(self.pg_config_get('ssl'), ssl)
+        eq_(self.config_get("ssl.sslmode"), sslmode)
+
+        expected_ciphers = (
+            "ECDHE-ECDSA-AES128-GCM-SHA256:"
+            "ECDHE-ECDSA-AES256-GCM-SHA384:"
+            "ECDHE-RSA-AES128-GCM-SHA256:"
+            "ECDHE-RSA-AES256-GCM-SHA384:"
+            "ECDHE-ECDSA-AES128-SHA256:"
+            "ECDHE-ECDSA-AES256-SHA384:"
+            "ECDHE-RSA-AES128-SHA256:"
+            "ECDHE-RSA-AES256-SHA384"
+        )
+
+        # TODO: also do this for monitor once we can have superuser access to
+        # the monitor
+        if not monitor:
+            eq_(self.pg_config_get("ssl_ciphers"), expected_ciphers)
+
+        def check_conn_string(conn_string):
+            print("checking connstring =", conn_string)
+            assert f"sslmode={sslmode}" in conn_string
+            if rootCert:
+                assert f"sslrootcert={rootCert}" in conn_string
+            if crl:
+                assert f"sslcrl={crl}" in conn_string
+
+        conn_string, _ = PGAutoCtl(self)\
+            .execute("show uri --monitor", 'show', 'uri', '--monitor')
```

Comment:
What about using `self.monitor_uri()` instead, to clean-up the code and make it easier to reuse?

Reply:
Done.